lib/ansible/modules/cloud/rpmostree/rpmostree_compose.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public licenses
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: rpmostree_compose
+short_description: Compose OSTree
+description:
+    - Entrypoint for tree composition, used on servers to prepare trees for replication by client systems
+version_added: "2.4"
+author: "Trishna Guha @trishnaguha"
+notes:
+    - Distribution should be Red Hat based (RHEL, CentOS, Fedora)
+requirements:
+    - rpm-ostree
+    - "python >= 2.6"
+options:
+    repo:
+        description:
+            - Path to OSTree repository
+        required: True
+        default: None
+    cachedir:
+        description:
+            - Cached state
+        required: False
+        default: None
+    manifest:
+        description:
+            - The manifest docker-json TREEFILE required for compose
+        required: True
+        default: None
+'''
+
+EXAMPLES = '''
+---
+# rpm-ostree compose
+- rpmostree_compose:
+    repo: /srv/repo
+    cachedir: /srv/cache
+    manifest: /path/to/fedora-atomic-docker-host.json
+'''
+
+RETURN = '''
+changed: [hostname]
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+class RPMOSTreeCompose(object):
+
+    def __init__(self, module):
+        self.module = module
+
+    def distribution(self):
+        """
+        Returns distribution RHEL/ CentOS/ Fedora <str>
+        """
+        with open('/etc/os-release', 'r') as release_file:
+            release_data = release_file.readlines()
```

Comment:
Read https://github.com/ansible/ansible/blob/devel/CODING_GUIDELINES.md#regexes. So you can ignore :)